### PR TITLE
Services & Brokers

### PR DESCRIPTION
**Pull request by mohammadhb**
feat: Add service broker
feat: Add Email Service
feat: Integrate Email Service into Auth
fix: Converted "Sync" Services to "Async" and Renamed "Sync" Services to Libs

### CODE REVIEW

File: src/lib/upload.go [hunk 1]
```diff
@@ -68,3 +57,15 @@ func Upload(file multipart.File, fileName string) (string, string) {
 
 	return fmt.Sprintf("%s/%s", config.Config.S3.CDNUrl, filename), fileName
 }
+
+// Initializing
+func InitS3Lib() {
```
Comment (jeyem) [inline src/lib/upload.go:62]: make this independent from other packages even config

File: src/services/mail.go [hunk 2]
```diff
@@ -1,69 +1,41 @@
...
-
-var SendGridClient SendGridType
+const EmailChannel = "email"
 
-type SendGridType struct {
-	ApiKey   string
-	Url      string
-	Disabled bool
+type EmailConfig struct {
+	Approach    string // [template|direct]
```
Comment (jeyem) [inline src/services/mail.go:12]: make constant and enum for all members like this , refer to models enums

**Comment by mohammadhb** [comment]
✅ Applied all of the change requests

**Review by jeyem — CHANGES_REQUESTED** [review]
CI&CD need to change to running consumers and related services

**Comment by mohammadhb** [comment]
✅ Done 
✅ @jeyem Also put all of the channels under shin/${channel} so that it socious-api wont confuse it with its cahnnels

**Review by jeyem — APPROVED** [review]

**Pull request merged**